From 22fd32e6c9b0cd5ed6fc513512838cf93328188d Mon Sep 17 00:00:00 2001
From: Alberto Ponces <ponces26@gmail.com>
Date: Thu, 29 Sep 2022 16:51:49 +0100
Subject: [PATCH 2/5] Revert "Additionally check le_set_event_mask command
 resturn status with UNSUPPORTED_LMP_OR_LL_PARAMETER"

This reverts commit bfddadac5e1c12ab3e032d15de210d6e4ccf2f3a.
---
 system/gd/hci/controller.cc | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/system/gd/hci/controller.cc b/system/gd/hci/controller.cc
index 8be21a20a3..da5986fcb7 100644
--- a/system/gd/hci/controller.cc
+++ b/system/gd/hci/controller.cc
@@ -540,7 +540,7 @@ struct Controller::impl {
   void le_set_event_mask(uint64_t le_event_mask) {
     std::unique_ptr<LeSetEventMaskBuilder> packet = LeSetEventMaskBuilder::Create(le_event_mask);
     hci_->EnqueueCommand(std::move(packet), module_.GetHandler()->BindOnceOn(
-                                                this, &Controller::impl::check_event_mask_status<LeSetEventMaskCompleteView>));
+                                                this, &Controller::impl::check_status<LeSetEventMaskCompleteView>));
   }
 
   template <class T>
@@ -551,15 +551,6 @@ struct Controller::impl {
     ASSERT(status_view.GetStatus() == ErrorCode::SUCCESS);
   }
 
-  template <class T>
-  void check_event_mask_status(CommandCompleteView view) {
-    ASSERT(view.IsValid());
-    auto status_view = T::Create(view);
-    ASSERT(status_view.IsValid());
-    ASSERT(status_view.GetStatus() == ErrorCode::SUCCESS ||
-           status_view.GetStatus() == ErrorCode::UNSUPPORTED_LMP_OR_LL_PARAMETER);
-  }
-
 #define OP_CODE_MAPPING(name)                                                  \
   case OpCode::name: {                                                         \
     uint16_t index = (uint16_t)OpCodeIndex::name;                              \
-- 
2.34.1

